### DISCUSSION TITLE
Vue2 项目组件测试，报两个TypeError(reading 'state' 和 reading '$on')

### DISCUSSION THREAD

**Discussion by zhengbangbo** [haixiangyan/jest-tutorial#34] [Q&A]
在跟随指南实践一次之后，我想在我的练习项目（Vue2）中尝试加入测试。

项目：https://github.com/zhengbangbo/sph-app/commit/5c275dc890848efa21d09a147e47fccb7ab46b79

无论我使用 [vue-testing-library](https://github.com/testing-library/vue-testing-library) 还是 [vue-test-utils](https://github.com/vuejs/vue-test-utils) 都会在 render 组件时报错，有两个

1. [Vue warn]: Error in render: "TypeError: Cannot read properties of undefined (reading 'state')"
2. [Vue warn]: Error in mounted hook: "TypeError: Cannot read properties of undefined (reading '$on')"

不知道各位有没有遇到过，是怎么解决的？

**Comment by ywj87132749** [comment]
不知道。。还在学习jest中。。。。。
